frontend/src/worksheets/WorkSheetOperationButtons.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { useHistory } from 'react-router-dom'
+import { PATH_VARIABLES } from '../Const'
+import EditButton from '../specialButtons/EditButton'
+import FinalizeButton from '../specialButtons/FinalizeButton'
+import CloseButton from '../specialButtons/CloseButton'
+import PdfButton from '../specialButtons/PdfButton'
+import workSheetPDF from './workSheetPDF'
+import useCurrentUser from '../hooks/useCurrentUser'
+import useWorkSheets from '../hooks/useWorkSheets'
+import { closeWorkSheet, finalizeWorkSheet } from '../api/WorkSheetAPI'
+
+function WorkSheetOperationButtons({ worksheet }) {
+  const { updateWorkSheets } = useWorkSheets()
```

Comment:
Ez nem szerencsés így, mert minden sor külön le fogja futtatni a useWorksheets dolgait. Ha van 5 munkalap, akkor legalább 5x le fog futni az API hívás. Az jó, hogy lett egy WorkSheetOperationButtons komponens, de a megoldás az lenne, hogy ez propsban kapja a szükséges adatokat és eseménykezelőket, a finalize és close pedig maradnak a lista komponensben.

Reply:
A munkalap generálás maradhat is, arra nem kell feltétlenül esemény és nem kell kint kezelni, mert annak nincsenek ilyen komoly mellékhatásai.

Reply:
Egyeztess Tibivel a másik csapatból, ő tud mesélni, hogy a hasonló megoldás milyen bugokat okozott náluk.

Reply:
Valóban sokszor lefutott. Javítottam.